src/scipp/core/dimensions.py

```diff
@@ -172,3 +174,33 @@ def _rename_dataset(
         scalar(0).broadcast(dims=ds.dims, shape=ds.shape), coords=ds.coords
     )
     return Dataset(coords=_rename_data_array(dummy, dims_dict).coords)
+
+
+_USED_AUX_DIMS: list[str] = []
+
+
+def new_dim_for(*data: Variable | DataArray) -> str:
+    """Return a dimension label that is not in the input's dimensions.
+
+    The returned label is intended for temporarily reshaping an array and should
+    not become visible to users.
+    The label is guaranteed to not be present in the input, but it may be used in
+    other variables.
```

Comment:
Maybe we should explain here why using uuid or other generated labels is problematic, i.e., why one should use this function?